### ISSUE TITLE
`cargo.buildScripts.*` are poor config names

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#13465]
Given `cargo.buildScripts.overrideCommand` allows an arbitrary command to be executed, these configs should not be prefixed with cargo. `buildScripts` is also a poor choice, given these are not build-script specific settings but also proc-macro related. We should come up with a better more fitting prefix, something like `compileTimeDeps.*`.

**Comment by bjorn3** [comment]
We depend on cargo's json message format for getting the out dir paths back. As well as to map these messages to the right rust-analyzer project model crate.

**Comment by Veykril** [comment]
Heh, that is a good point, but we do the same for diagnostics right? Though I guess those are rustc json blobs so technically not cargo related.

I guess the `cargo` stays, `buildScripts` is still less than ideal, iirc there were some people confused as to why those settings are related to proc-macros.

**Comment by bjorn3** [comment]
> Heh, that is a good point, but we do the same for diagnostics right? Though I guess those are rustc json blobs so technically not cargo related.

Indeed! Rust-analyzer has a strong dependency on rustc which is unlikely to change any time soon, but it isn't as much dependent on cargo due to rust-project.json. I think we should at least document which json messages rust-analyzer expects from cargo or another build system emulating cargo's build messages.
